backend/app/model/mixins/relationships.rb

```diff
@@ -92,58 +92,123 @@ def self.transfer(target, victims)
+                  # relationship so we'll skip over it.  But while we're here, make
+                  # sure we're not about to create a circular relationship.
 
-            if relationship[target_col]
-              # This column is already used to reference the other record in our
-              # relationship so we'll skip over it.  But while we're here, make
-              # sure we're not about to create a circular relationship.
+                  if relationship[target_col] == target.id
+                    raise "Transfer would create a circular relationship!"
+                  end
+
+                elsif relationship.is_a?(Relationships::SubContainerTopContainerLink)
+                  identify_duplicate_containers(target, relationship, target_col, dups)
+
+                else
+                  # Found a free column.  Store our updated reference here.
+                  !target_pre.empty? && target_pre.first[parent_col] == relationship[parent_col] ? dups << relationship : relationship[target_col] = target.id
+                  break
+                end
 
-              if relationship[target_col] == target.id
-                raise "Transfer would create a circular relationship!"
               end
 
-            else
-              # Found a free column.  Store our updated reference here.
-              relationship[target_col] = target.id
-              break
-            end
+              relationship[:system_mtime] = Time.now
+              relationship[:user_mtime] = Time.now
+
+              relationship.save
 
+              cleanup_duplicates(dups)
+            end
           end
+        end
+      end
+    end
+  end
+
+
+  # If we're merging a record of type A with relationship R into a record of
+  # type B, type B must also support that relationship type.  If it doesn't,
+  # we risk losing data through the merge and should abort.
+  def self.confirm_accepts_target(victim_model, vics)
+    unless participating_models.include?(victim_model)
+      found = self.find_by_participant_ids(victim_model, vics.map(&:id))
+
+      unless found.empty?
+        raise ReferenceError.new("#{victim_model} to be merged has data for relationship #{self}, but target record doesn't support it.")
+      end
+    end
+  end
+
 
-          relationship[:system_mtime] = Time.now
-          relationship[:user_mtime] = Time.now
+  # ANW-952: After merging objects together you may be left with two
+  # relationships that link to the same post-merge record.  This deletes
+  # those duplicated relationships after the merge process is complete.
+  def self.cleanup_duplicates(dups)
+    if !dups.empty?
+      dups.each {|d| d.delete}
+    end
+  end
 
-          relationship.save
+
+  def self.identify_duplicate_containers(target, relationship, target_col, dups)
+    target_relationships = find_by_participant(target)
+    target_relationships.each do |target_relationship|
```

Comment:
Same as above, can be reduced to: `find_by_participant(target).each do |target_relationship|`